cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -86,33 +90,49 @@ Location getParent(Location location) {
   }
 
   /**
-   * Deletes dir if it is empty, and recursively deletes parent dirs if they are empty too. The recursion stops at
-   * non-empty parent or base directory. If dir is not child of base directory then the recursion stops at root.
-   * @param dir dir to be deleted.
+   * For the specified directory to be deleted, finds its namespaced log location, then deletes
+   * @param dir
+   * @throws IOException
    */
-  void deleteEmptyDir(Location dir) {
+  void deleteEmptyDir(Location dir) throws IOException {
     LOG.debug("Got path {}", dir.toURI());
 
+    List<Location> namespaceDirs = rootDir.list();
```

Comment:
Will have to do some sort of regex matching then, right? Or something with ``getParent(location)``. Explored both, seemed a little too hacky for comfort to me for a delete operation.